core/src/main/java/com/sequenceiq/cloudbreak/controller/StackCreatorService.java

```diff
@@ -379,9 +381,43 @@ void fillInstanceMetadata(DetailedEnvironmentResponse environment, Stack stack)
                 instanceMetaData.setPrivateId(privateIdNumber++);
                 instanceMetaData.setInstanceStatus(InstanceStatus.REQUESTED);
             }
-            multiAzCalculatorService.calculateByRoundRobin(subnetAzPairs, instanceGroup);
-            prepareInstanceMetaDataSubnetAndAvailabilityZoneAndRackId(stackSubnetId, stackAz, instanceGroup);
         }
+        if (stack.getType() == StackType.DATALAKE) {
+            for (InstanceGroup instanceGroup : sortInstanceGroups(stack)) {
+                if (!(instanceGroup.getGroupName().equals("GATEWAY") || instanceGroup.getGroupName().equals("AUXILIARY"))) {
```

Comment:
I guess we should use a constant here or https://github.com/hortonworks/cloudbreak/pull/14741/files#diff-9f31a0a3c9008721bb5c389d6812fbfea2f7f45a45fc0e0ca57903b9affbf224R7

Reply:
addressed